src/components/deploymentConfig/DeploymentConfig.tsx

```diff
@@ -297,6 +284,8 @@ export default function DeploymentConfig({
             },
         }
 
+        handleTabSelection(isApprovalPending ? 2 : 3)
```

Comment:
Should this happen post dispatch?

Reply:
We can't move this after the payload since we need to be in Advanced yaml mode before the dispatch fires